plugins/inputs/knx_listener/knx_listener.go

```diff
@@ -0,0 +1,183 @@
+package knx_listener
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/vapourismo/knx-go/knx"
+	"github.com/vapourismo/knx-go/knx/dpt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type KNXInterface interface {
+	Inbound() <-chan knx.GroupEvent
+	Close()
+}
+
+type addressTarget struct {
+	measurement string
+	datapoint   dpt.DatapointValue
+}
+
+type Measurement struct {
+	Name      string
+	Dpt       string
+	Addresses []string
+}
+
+type KNXListener struct {
+	ServiceType    string        `toml:"service_type"`
+	ServiceAddress string        `toml:"service_address"`
+	Measurements   []Measurement `toml:"measurement"`
+
+	client      KNXInterface
+	gaTargetMap map[string]addressTarget
+
+	acc telegraf.Accumulator
+}
+
+func (kl *KNXListener) Description() string {
+	return "Listener capable of handling KNX bus messages provided through a KNX-IP Interface."
+}
+
+func (kl *KNXListener) SampleConfig() string {
+	return `
+  # Type of KNX-IP interface.
+  # Can be either "tunnel" or "router".
+  service_type = "tunnel"
+
+  # Address of the KNX-IP interface.
+  service_address = "localhost:3671"
+
+  ## Measurement definition(s)
+  # [[inputs.KNXListener.measurement]]
+  #   # Name of the measurement
+  #   name = "temperature"
+  #   # Datapoint-Type (DPT) of the KNX messages
+  #   dpt = "9.001"
+  #   # List of Group-Addresses (GAs) assigned to the measurement
+  #   addresses = ["5/5/1"]
+
+  # [[inputs.KNXListener.measurement]]
+  #   name = "illumination"
+  #   dpt = "9.004"
+  #   addresses = ["5/5/3"]
+`
+}
+
+func (kl *KNXListener) Gather(acc telegraf.Accumulator) error {
+	return nil
+}
+
+func (kl *KNXListener) Start(acc telegraf.Accumulator) error {
+	// Store the accumulator for later use
+	kl.acc = acc
+
+	// Construct the mapping of Group-addresses (GAs) to DPTs and the name
+	// of the measurement
+	kl.gaTargetMap = make(map[string]addressTarget)
+	for _, m := range kl.Measurements {
+		log.Printf("D! [inputs.KNXListener] group-address mapping for measurement \"%s\"", m.Name)
```

Comment:
If you want to log, you should declare the Telegraf logger, not use the log package -> https://github.com/influxdata/telegraf/wiki/Review#reviewing-plugin-code
(for all logs)